### ISSUE TITLE
dartLangSpec.tex says that static must precede metadata

### ISSUE THREAD

**Issue by peter-ahe-google** [dart-lang/sdk#29614]
```
{\bf methodSignature:}constructorSignature initializers?;
      factoryConstructorSignature;
      \STATIC{}? functionSignature;
      \STATIC{}? getterSignature;
      \STATIC{}? setterSignature;
      operatorSignature
    .
```

BUT:

```
{\bf functionSignature:}
    metadata returnType? identifier formalParameterList
    .
```

So according to the grammar, this should be correct:

```
static @override foo() => 42
```

And this isn't correct:

```
@override static foo() => 42
```

**Comment by eernstg** [comment]
Right, that is a fault in the grammar. It is already covered in https://codereview.chromium.org/2688903004, and I'll rework the grammar in the spec to include corrections made there.

**Comment by eernstg** [comment]
No milestone now: This correction is part of a larger update of grammar related parts of the specification, based on [Dart.g](https://github.com/dart-lang/sdk/blob/master/docs/language/Dart.g).

**Comment by eernstg** [comment]
Closing: Fixed in ea7cf292082d2648f1b218d3c92d3b23b86a5d65.
